libraries/core_libs/consensus/src/pbft/pbft_manager.cpp

```diff
@@ -1634,13 +1634,16 @@ void PbftManager::finalize_(SyncBlock &&sync_block, std::vector<h256> &&finalize
         }
 
         db_->addProposalPeriodDagLevelsMapToBatch(anchor->getLevel() + max_levels_per_period_, period, batch);
+        ptr->finalized();
```

Comment:
maybe it would be better to use `block_applying_emitter_` as it run on separate thread and it will do the same work. This way we are not slowing down finalization process